integration-tests/subscriptions.test.ts

```diff
@@ -31,44 +27,81 @@ test('empty test placeholder', () => {
+    if (!SUBSCRIPTIONS_ENDPOINT) {
+        throw new Error('SUBSCRIPTIONS_ENDPOINT environment variable is not defined');
+    }
+    let client: AxiosInstance;
+
+    const subscriptionResource = {
+        resourceType: 'Subscription',
+        status: 'requested',
+        // get a time 10 seconds (10000 ms) in the future
+        end: new Date(new Date().getTime() + 10000).toISOString(),
+        reason: 'Monitor Patients with name Smith',
+        criteria: 'Patient?name=Smith',
+        channel: {
+            type: 'rest-hook',
+            endpoint: SUBSCRIPTIONS_ENDPOINT!,
+            payload: 'application/fhir+json',
+            header: [`x-api-key: ${process.env.SUBSCRIPTIONS_API_KEY}`],
+        },
+    };
+
     describe('FHIR Subscriptions', () => {
         let subscriptionsHelper: SubscriptionsHelper;
 
-        beforeAll(() => {
+        beforeAll(async () => {
             if (SUBSCRIPTIONS_NOTIFICATIONS_TABLE === undefined) {
                 throw new Error('SUBSCRIPTIONS_NOTIFICATIONS_TABLE environment variable is not defined');
             }
             subscriptionsHelper = new SubscriptionsHelper(SUBSCRIPTIONS_NOTIFICATIONS_TABLE);
+            client = await getFhirClient();
         });
 
         test('test', async () => {
             const x = await subscriptionsHelper.getNotifications('/lala');
             console.log(x);
         });
+
+        if (process.env.MULTI_TENANCY_ENABLED === 'true') {
+            test('tenant isolation', async () => {
+                const clientAnotherTenant = await getFhirClient({ tenant: 'tenant2' });
+                // tenant 1 creates a subscription
+                const subResource = clone(subscriptionResource);
+                // make sure the end date isn't caught by the reaper before the test completes
+                subResource.end = new Date(new Date().getTime() + 100000).toISOString();
+                const postResult = await client.post('Subscription', subscriptionResource);
+                expect(postResult.status).toEqual(201);
+                const resourceThatMatchesSubscription = {
+                    resourceType: 'Patient',
+                    name: [
+                        {
+                            given: ['Smith'],
+                            family: 'Smith',
+                        },
+                    ],
+                };
+                // post matching resource on another tenant
+                const postPatientResult = await clientAnotherTenant.post('Patient', resourceThatMatchesSubscription);
+                expect(postPatientResult.status).toEqual(201);
+                // give SLA of 20 seconds for notification to be placed in ddb table
+                await new Promise((r) => setTimeout(r, 20000));
+                // make sure no notification was receieved for first tenant
+                const notifications = await subscriptionsHelper.getNotifications(
+                    `/Patient/${postPatientResult.data.id}`,
```

Comment:
This can get mixed up with notifications from other tests. 

I suggest you add a UUID to the endpoint in all test Subscriptions, so that you can lookup in DDB for the path with the exact same UUID.

```ts
endpoint: `${SUBSCRIPTIONS_ENDPOINT}/<uuid>`
```